File: convalida-validators/src/test/java/convalida/validators/ConfirmPasswordValidatorTest.java
package convalida.validators;

import android.text.Editable;
import android.widget.EditText;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * @author Wellington Costa on 02/11/2017.
 */
public class ConfirmPasswordValidatorTest extends BaseTest {

    private EditText passwordEditText;

    @Before public void setUpValidators() {
        passwordEditText = mock(EditText.class);
        when(passwordEditText.getText()).thenReturn(mock(Editable.class));
    }

    @Test public void emptyValues() {
        ConfirmEmailValidator validator = new ConfirmEmailValidator(passwordEditText, mockEditText, errorMessage, true);
        when(passwordEditText.getText().toString()).thenReturn("");
        when(mockEditText.getText().toString()).thenReturn("");
        assertEquals(validator.validate(), true);
    }

    @Test public void passwordsDoesNotMatch() {
        ConfirmPasswordValidator validator = new ConfirmPasswordValidator(passwordEditText, mockEditText, errorMessage, true);
        when(passwordEditText.getText().toString()).thenReturn("test@123");
        when(mockEditText.getText().toString()).thenReturn("test@12");
        assertEquals(validator.validate(), false);
    }

    @Test public void passwordsMatch() {
        ConfirmPasswordValidator validatorWithEditText = new ConfirmPasswordValidator(passwordEditText, mockEditText, errorMessage, true);
        when(passwordEditText.getText().toString()).thenReturn("test@123");
        when(mockEditText.getText().toString()).thenReturn("test@123");
        assertEquals(validatorWithEditText.validate(), true);
    }

}

File: convalida-validators/src/test/java/convalida/validators/ConfirmPasswordValidatorTest.kt
package convalida.validators

import android.support.design.widget.TextInputLayout
import android.text.Editable
import android.widget.EditText

import org.junit.Before
import org.junit.Test

import org.junit.Assert.*
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`

/**
 * @author Wellington Costa on 02/11/2017.
 */
class ConfirmPasswordValidatorTest : BaseTest() {

    private lateinit var passwordEditText: EditText

    @Before
    fun setUpValidators() {
        passwordEditText = mock(EditText::class.java)
        `when`(passwordEditText.text).thenReturn(mock(Editable::class.java))
    }

    @Test
    fun passwordsDoesNotMatch() {
        val validator = ConfirmPasswordValidator(passwordEditText, mockEditText, errorMessage)
        `when`(passwordEditText.text.toString()).thenReturn("test@123")
        `when`(mockEditText.text.toString()).thenReturn("test@12")
        assertEquals(validator.validate(), false)
    }

    @Test
    fun passwordsMatch() {
        val validatorWithEditText = ConfirmPasswordValidator(passwordEditText, mockEditText, errorMessage)
        `when`(passwordEditText.text.toString()).thenReturn("test@123")
        `when`(mockEditText.getText().toString()).thenReturn("test@123")
        assertEquals(validatorWithEditText.validate(), true)

    }

}